<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>APRICOT by malvikasharan</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">APRICOT</h1>
      <h2 class="project-tagline">A computational pipeline to identify specific functional classes of proteins.</h2>
      <a href="https://github.com/malvikasharan/APRICOT" class="btn">View on GitHub</a>
      <a href="https://github.com/malvikasharan/APRICOT/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/malvikasharan/APRICOT/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h3>
<a id="APRICOT" class="anchor" href="#apricot" aria-hidden="true"><span class="octicon octicon-link"></span></a>APRICOT</h3>

<p>APRICOT is a computational pipeline for the identification of functional features of interest in large protein sets. For the characterization of user-provided query proteins with specific functional characteristics, APRICOT uses efficient sequence-based algorithms and predictive models like signature motifs of protein families. 
Due to the flexible choice of reference predictive models, APRICOT has the potential to  the unexplored functions of proteins with a diversity of binding capabilities.</p>

<h3>
<a id="Introduction" class="anchor" href="#introduction" aria-hidden="true"><span class="octicon octicon-link"></span></a>INTRODUCTION</h3>

<p>The initial focus of this project was to identify functional domains in bacterial proteins that have the potential to interact with RNA and understand their regulatory roles and mechanisms. Therefore, this tool was named as <strong>APRICOT</strong> (stands for <strong>A</strong>nalysing <strong>P</strong>rotein <strong>R</strong>NA <strong>I</strong>nteractions by <strong>Co</strong>mputational <strong>T</strong>echniques). Using this pipeline, a proteome-wide screening of RBPs was carried out in Salmonella Typhimurium. Candidate RNA-interacting proteins were selected and tested in co-immunoprecipitation and high throughput RNA sequencing based assays (RIP-seq) by experimentalists in our lab to validate their RNA-binding potential and identify bound RNAs. The statistical analysis of the sequencing data indicated a significant number of them as putative RBPs and revealed the distinct expression patterns for the proteins that bind to several RNAs compared to the proteins that have fewer RNA partners.</p>

<h3>
<a id="authors-and-contributors" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span class="octicon octicon-link"></span></a>Authors and Contributors</h3>

<p>The tool is designed and developed by Malvika Sharan <a href="https://github.com/malvikasharan" class="user-mention">@malvikasharan</a> in the lab of <a href="http://www.imib-wuerzburg.de/research/vogel/group-leader/">Prof. Dr. Jörg Vogel</a> in the Institute for Molecular Infection Biology at the University of Würzburg. Dr. Konrad Förstner <a href="https://github.com/konrad" class="user-mention">@konrad</a> contributed to the project by providing important technical supervision and discussions. The authors thank the members of Vogel lab, especially <a href="http://www.imib-wuerzburg.de/de/home/mitarbeiter/?tx_wecstaffdirectory_pi1%5Bcurstaff%5D=322&amp;cHash=206fb115ed7f1e7a469007d2c40baae7">Dr. Charlotte Michaux</a> and <a href="http://www.imib-wuerzburg.de/de/students/info/doktorandenkolleg/doctoral-projects/caroline-tawk/">Caroline Tawk</a> for their biological point of views and important contributions in the related projects.</p>

<h3>
<a id="support-or-contact" class="anchor" href="#support-or-contact" aria-hidden="true"><span class="octicon octicon-link"></span></a>Support or Contact</h3>

<p>Having trouble with Pages? Check out our <a href="https://help.github.com/APRICOT/documentation">documentation</a> or <a href="malvikasharan@gmail.com">contact us</a> and we’ll help you sort it out.</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/malvikasharan/APRICOT">APRICOT</a> is maintained by <a href="https://github.com/malvikasharan">malvikasharan</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

            <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-68940079-1");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>
