##APRICOT
A tool to explore functional classes of proteins.

**APRICOT** is a computational pipeline for the identification of specific functional classes of interest in large protein sets. The pipeline uses efficient sequence-based algorithms and predictive models like signature motifs of protein families for the characterization of user-provided query proteins with specific functional features. The dynamic framework of APRICOT allows the identification of unexplored functional classes of interest in the large protein sets or the entire proteome.

###Introduction

The initial focus of this project was to identify functional domains in bacterial proteins that have the potential to interact with RNA and understand their regulatory roles and mechanisms, therefore it was named as **APRICOT** (stands for **A**nalysing **P**rotein **R**NA **I**nteractions by **Co**mbined-scoring **T**echniques). Using this pipeline, a proteome-wide screening of RBPs was carried out in Salmonella Typhimurium. Candidate RNA-binding proteins (RBPs) were selected and subjected to co-immunoprecipitation and high throughput RNA sequencing based assays (RIP-seq) in our lab to validate their RNA-binding potential and identify bound RNAs. The statistical analysis of the sequencing data indicated a significant number of them as putative RBPs and revealed the distinct expression patterns for the proteins that bind to several RNAs compared to the proteins that have fewer RNA partners.

Due to the flexible choice of reference predictive models, several proteins of known classes are being tested with this pipeline. The result obtained from the experimental validations were taken into account to improve the functional characterization of the query proteins by APRICOT.  

###Authors and Contributors

The tool is designed and developed by Malvika Sharan @malvikasharan in the lab of Prof. Dr. Jörg Vogel in the Institute for Molecular Infection Biology at the University of Würzburg. Dr. Konrad Förstner @konrad contributed to the project by providing important technical supervision and discussions. The authors are grateful to Prof. Thomas dandekar, Dr. Ana Eulalio, Dr. Charlotte Michaux and Caroline Taouk for critical discussions and feedback.

###Documentation

The documentation is under preparation.


